jsonchecker/__init__.py

```diff
@@ -91,6 +81,71 @@ def exit(self):
             return 0
 
 
+class DuplicateKeyFinder(DuplicateFinder):
+    """Duplicate Key Finder."""
+
+    def checker(self, seq, fname):
+        """Check routine."""
+        d = {}
+        for key, value in seq:
+            if key in d:
+                self.mark_error(key, fname)
+            else:
+                d[key] = value
+        return d
+
+    def exit(self):
+        """print errors and exit."""
+        return self.exit_with_msg('keys')
+
+
+class DuplicateValueFinder(DuplicateFinder):
+    """Duplicate Value Finder."""
+
+    def hashable(self, v):
+        """Determine whether `v` can be hashed."""
+        try:
+            hash(v)
+        except TypeError:
+            return False
+        return True
+
+    def dupes_in_list(self, l):
+        """Return hashable duplicates from this list."""
+        seen = set()
```

Comment:
you could use `collections.Counter` for the seen lists, especially if it useful to know how many times a value appears, such as ordering them errors by number of occurrences.

Sorry I forgot to mention this earlier.  But it isnt a significant issue if you dont have time to attempt it.  I'll merge this tomorrow either way.